addons/mass_mailing/tests/test_mailing_mailing_schedule_date.py

```diff
@@ -31,3 +31,77 @@ def test_mailing_schedule_date(self):
+        mailing = self.env['mailing.mailing'].create({
+            'name': 'mailing',
+            'subject': 'some subject',
+            'mailing_model_id': self.env['ir.model']._get('res.partner').id,
+            'state' : 'draft'
+        })
+
+        cron_job = self.env.ref('mass_mailing.ir_cron_mass_mailing_queue').sudo()
+        cron_job_nextcall = datetime(2099, 4, 30, 9, 0) # way in the future
+        cron_job.write({'nextcall' : cron_job_nextcall})
+
+        mailing.action_launch()
+        end_time = datetime.now() + relativedelta(seconds=30)
+        test_now_schedule_date = True if start_time <= mailing.next_departure <= end_time else False
+        # assert that the schedule_date and schedule_type fields are correct and that the mailing is put in queue
+        self.assertEqual(test_now_schedule_date, True)
+        self.assertIsNot(mailing.schedule_date, cron_job_nextcall)
+        self.assertEqual(mailing.schedule_type, 'now')
+        self.assertEqual(mailing.state, 'in_queue')
+
+        # test cancel button
+        mailing.action_cancel()
+
+        # assert if cancel reset fields as expected
+        self.assertEqual(mailing.state, 'draft')
+        self.assertEqual(mailing.schedule_type, 'now')
+        self.assertEqual(mailing.schedule_date, False)
+        self.assertEqual(mailing.next_departure, False)
+
+        # case where client uses schedule wizard to chose a date between now and cron.job nextcall
+        # create a schedule date wizard
+        wizard_form = Form(
+            self.env['mailing.mailing.schedule.date'].with_context(default_mass_mailing_id=mailing.id))
+
+        # set a schedule date in near future
+        schedule_date_user = datetime.now() + relativedelta(minutes=50)
+        wizard_form.schedule_date = schedule_date_user
+        wizard = wizard_form.save()
+        wizard.action_schedule_date()
+
+        self.assertEqual(mailing.schedule_date, schedule_date_user)
+        self.assertEqual(mailing.next_departure, schedule_date_user)
+        self.assertEqual(mailing.schedule_type, 'scheduled')
+        self.assertEqual(mailing.state, 'in_queue')
+
+        # case where client uses schedule wizard to chose a date after cron.job nextcall
+        # which means mails will get send after that date
+
+        mailing.action_cancel()
+        # # create a schedule date wizard
+        wizard_form = Form(
+            self.env['mailing.mailing.schedule.date'].with_context(default_mass_mailing_id=mailing.id))
+
+        # set a schedule date after cron_job_nextcall
+        schedule_date_user = cron_job_nextcall + relativedelta(days=50)
+        wizard_form.schedule_date = schedule_date_user
+        wizard = wizard_form.save()
+        wizard.action_schedule_date()
+
+        # assert if schedule_date/departure is after cron_job_nextcall
+        test_after_cron_job_time = mailing.next_departure > cron_job_nextcall
+        self.assertEqual(test_after_cron_job_time, True)
```

Comment:
`assertTrue` exists ^^
(same above)

Reply:
Tests are new for me, thanks for teaching me new stuff ;)